src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1101 @@
+                return &storage;
+            }
+        }
+        return nullptr;
+    }
+
+    bool InPool() const
+    {
+        constexpr auto end = pool + ArraySize(pool);
+        return pool <= this && this < end;
+    }
+
+public:
+    EntryStorage() = default;
+
+    void Init()
+    {
+        if (!mInUse)
+        {
+            Clear();
+            mInUse = true;
+        }
+    }
+
+    bool InUse() const { return mInUse; }
+
+    void Release()
+    {
+        if (InPool())
+        {
+            mInUse = false;
+        }
+    }
+
+    void Clear()
+    {
+        mInUse = false;
+        mFabricIndex = chip::kUndefinedFabricIndex;
+        mAuthMode = AuthMode::kPase;
+        mPrivilege = Privilege::kView;
+        memset(mSubjects, SubjectStorage::kEmptyByte, sizeof(mSubjects));
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+        memset(mTargets, TargetStorage::kEmptyByte, sizeof(mTargets));
+#pragma GCC diagnostic pop
+#else
+        for (auto & target : mTargets)
+        {
+            target.Clear();
+        }
+#endif
+    }
+
+private:
+    static size_t FilterIndex(size_t index, const FabricIndex * fabricIndex)
+    {
+        if (fabricIndex != nullptr)
+        {
+            size_t unfiltered = 0;
+            size_t filtered = 0;
+            for (auto & storage : acl)
+            {
+                if (storage.InUse())
+                {
+                    return ArraySize(acl);
+                }
+                if (storage.mFabricIndex == *fabricIndex)
+                {
+                    if (filtered == index)
+                    {
+                        break;
+                    }
+                    filtered++;
+                }
+                unfiltered++;
+            }
+            return unfiltered;
+        }
+        return index;
+    }
+
+public:
+    static constexpr int kMaxSubjects = 4;
+    static constexpr int kMaxTargets = 3;
+
+private:
+    bool mInUse;
+
+public:
+    FabricIndex mFabricIndex;
+    AuthMode mAuthMode;
+    Privilege mPrivilege;
+    SubjectStorage mSubjects[kMaxSubjects];
+    TargetStorage mTargets[kMaxTargets];
+};
+
+class EntryDelegate : public Entry::Delegate
+{
+public:
+    // Pool support
+    static EntryDelegate pool[kEntryDelegatePoolSize];
+
+    static EntryDelegate * Find(Entry::Delegate & candidate)
+    {
+        if (InPool(candidate))
+        {
+            return &static_cast<EntryDelegate &>(candidate);
+        }
+        for (auto & delegate : pool)
+        {
+            if (!delegate.InUse())
+            {
+                return &delegate;
+            }
+        }
+        return nullptr;
+    }
+
+    static bool InPool(const Entry::Delegate & delegate)
+    {
+        constexpr auto end = pool + ArraySize(pool);
+        return pool <= &delegate && &delegate < end;
+    }
+
+public:
+    void Release() override
+    {
+        assert(mStorage != nullptr);
```

Comment:
VerifyOrDie is recommended here for convention overall.

Reply:
This conversation should be resolvable now.